src/punycode.cpp

```diff
@@ -13,9 +13,20 @@ constexpr int32_t initial_bias = 72;
 constexpr uint32_t initial_n = 128;
 
 static constexpr int32_t char_to_digit_value(char value) {
```

Comment:
Though it is not explicit, `char_to_digit_value` is only called on lower case. It is used right after `ascii_map` which itself does the mapping to lower case ASCII.